PAY-6182: Adapt docs to Payment Services drop-in V2

Tomasito665 opened the pull request:
## Purpose of this pull request

Update Payment Services documentation to include breaking drop-in initialization changes and new Apple Pay container, which is currently available in NPM as a V2-alpha release: https://www.npmjs.com/package/@dropins/storefront-payment-services/v/2.0.0-alpha1.

## Changes

Main changes
- Add `containers/apple-pay/` page to document new apple pay container
- Add `initialization/` page to document new init config & method availability events
- Update installation page to include apple pay domain verification instructions
- Remove `apiUrl` and `getCustomerToken` props from credit card container page
- Add apple pay to payment services overview page (alongside credit card)

Boy-scouting
- Remove unused import (CodeImport) from dictionary.mdx

### Associated JIRA ticket

https://jira.corp.adobe.com/browse/PAY-6182


### Staging preview

Not available.


## Affected pages

- https://experienceleague.adobe.com/developer/commerce/storefront/dropins/payment-services/
- https://experienceleague.adobe.com/developer/commerce/storefront/dropins/payment-services/installation/
- https://experienceleague.adobe.com/developer/commerce/storefront/dropins/payment-services/containers/credit-card/
- https://experienceleague.adobe.com/developer/commerce/storefront/dropins/payment-services/dictionary/ *

_\* will be updated automatically when we release the public version of the npm package_


## Links to source code

- https://github.com/adobe-commerce/storefront-payment-services/pull/5
- https://github.com/adobe-commerce/storefront-payment-services/pull/7
- https://github.com/adobe-commerce/storefront-payment-services/pull/8

Commit:
PAY-6182: Adapt docs to Payment Services drop-in V2

Main changes
- Add `containers/apple-pay/` page to document new apple pay container
- Add `initialization/` page to document new init config & method availability events
- Update installation page to include apple pay domain verification instructions
- Remove `apiUrl` and `getCustomerToken` props from credit card container page
- Add apple pay to payment services overview page (alongside credit card)

Boy-scouting
- Remove unused import (CodeImport) from dictionary.mdx

bdenham commented on src/content/docs/dropins/payment-services/index.mdx:
```diff
@@ -3,16 +3,15 @@ title: Payment Services overview
-import List from '@components/List.astro';
-
-The Payment Services drop-in component renders the credit card form, allowing shoppers to efficiently enter their payment details. The drop-in currently supports credit/debit cards only.
+The Payment Services drop-in component renders the credit card form and the apple pay button.
```
```suggestion
The Payment Services drop-in component renders the credit card form and the Apple pay button.
```

bdenham commented on src/content/docs/dropins/payment-services/index.mdx:
```diff
@@ -3,16 +3,15 @@ title: Payment Services overview
-The Payment Services drop-in component uses one container, `CreditCard`:
+The Payment Services drop-in component provides two containers:
+
+- **Credit card container**: the [`CreditCard`](/dropins/payment-services/containers/credit-card) container renders a form that shoppers can enter their card details in to place an order using a credit or debit card.
```
```suggestion
- **Credit card container:** The `CreditCard` container renders a form where shoppers enter their card details to place an order with a credit or debit card.
```

danidelcar replied:
@bdenham why did we remove the link from this entry?

bdenham commented on src/content/docs/dropins/payment-services/index.mdx:
```diff
@@ -3,16 +3,15 @@ title: Payment Services overview
+- **Credit card container**: the [`CreditCard`](/dropins/payment-services/containers/credit-card) container renders a form that shoppers can enter their card details in to place an order using a credit or debit card.
 
-- **Credit card container**: The [`CreditCard`](/dropins/payment-services/containers/credit-card) container is designed to allow placing an order with the credit card details during the checkout process.
+- **Apple pay container**: the [`ApplePay`](/dropins/payment-services/containers/apple-pay) container renders an Apple Pay button that shoppers can use on Apple devices to place an order using Apple Pay.
```
```suggestion
- **Apple Pay container:** The `ApplePay` container renders an Apple Pay button that shoppers on Apple devices can use to place an order.
```

danidelcar replied:
@bdenham why did we remove the link from this entry?

bdenham commented on src/content/docs/dropins/payment-services/initialization.mdx:
```diff
@@ -0,0 +1,43 @@
+
+## Initialization is asynchronous
+
+The Payment Services drop-in initializer is completely asynchronous. Even if mounted immediately, the drop-in will initialize in the background and notify initialization updates through custom events.
```
```suggestion
The Payment Services drop-in initializer is fully asynchronous. Even if mounted immediately, it initializes in the background and emits initialization updates via custom events.
```

bdenham commented on src/content/docs/dropins/payment-services/initialization.mdx:
```diff
@@ -0,0 +1,43 @@
+
+## Payment method availability
+
+The Payment Services drop-in will emit a custom `payment-services/method-available` event for every payment method that is determined to be available during the asynchronous initialization process. The event payload is a single string containing the payment method code.
```
```suggestion
During asynchronous initialization, the Payment Services drop-in emits a `payment-services/method-available` event for each available payment method. The event payload is a single string containing the payment method code.
```

bdenham commented on src/content/docs/dropins/payment-services/installation.mdx:
```diff
@@ -3,16 +3,53 @@ title: Payment Services installation
+import Tasks from '@components/Tasks.astro';
 
-Before you can use the Payment Services component on your storefront, you must [configure the Services Connector](https://experienceleague.adobe.com/en/docs/commerce-learn/tutorials/admin/adobe-commerce-services/configure-adobe-commerce-services-connector) and [onboard](https://experienceleague.adobe.com/en/docs/commerce/payment-services/get-started/onboard) the Payment Services extension in the  ...
+This guide will help you install and configure the Payment Services drop-in component in your storefront.
```
```suggestion
This guide explains how to install and configure the Payment Services drop-in component in your storefront.
```

bdenham commented on src/content/docs/dropins/payment-services/installation.mdx:
```diff
@@ -3,16 +3,53 @@ title: Payment Services installation
-The Payment Services drop-in component requires the [Payment Services extension](https://experienceleague.adobe.com/en/docs/commerce/payment-services/release-notes#v2100) version 2.10.0 or higher.
-:::
+## Payment Services extension <Badge tooltip="Applies to Adobe Commerce on Cloud projects (Adobe-managed PaaS infrastructure) and on-premises projects only." text="PaaS only" variant="note" href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/product-solutions" />
+The Payment Services drop-in component requires version <Link href="https://experienceleague.adobe.com/en/docs/commerce/payment-services/release-notes#v2100" text="2.10.0" /> or higher of the Payment Services extension.
```
```suggestion
The Payment Services drop-in requires the Payment Services extension version <Link href="https://experienceleague.adobe.com/en/docs/commerce/payment-services/release-notes#v2100" text="2.10.0" /> or higher.
```

bdenham commented on src/content/docs/dropins/payment-services/installation.mdx:
```diff
@@ -3,16 +3,53 @@ title: Payment Services installation
 
-## Installation
+## Onboard to Payment Services
+Before you can use the Payment Services component on your storefront, you must <Link href="https://experienceleague.adobe.com/en/docs/commerce/payment-services/get-started/onboard" text="onboard" /> to Payment Services in the Adobe Commerce Admin.
```
```suggestion
Before you can use the Payment Services component in your storefront, you must <Link href="https://experienceleague.adobe.com/en/docs/commerce/payment-services/get-started/onboard" text="onboard" /> to Payment Services in the Adobe Commerce Admin.
```

bdenham commented on src/content/docs/dropins/payment-services/installation.mdx:
```diff
@@ -3,16 +3,53 @@ title: Payment Services installation
 
-The Payment Services component relies on containers from several other drop-in components, you must install and configure those components before you can use the Payment Services component to place an order.
+## Apple Pay domain verification file <Badge tooltip="Applies to Adobe Commerce as a Cloud Service and Adobe Commerce Optimizer projects only (Adobe-managed SaaS infrastructure)." text="SaaS only" variant="success" href="https://experienceleague.adobe.com/en/docs/commerce/user-guides/product-solutions" />
+The following steps show how to register your storefront domain with Apple, which is required for using Apple Pay.
```
```suggestion
The following steps explain how to register your storefront domain with Apple, which is required to use Apple Pay.
```

bdenham commented on src/content/docs/dropins/payment-services/installation.mdx:
```diff
@@ -3,16 +3,53 @@ title: Payment Services installation
+
+  <Task>
+    ### Serve as binary content
+    Add the file to the following path in your storefront repository. Note the added `.bin` extension, which (as of today) is the only way we have in EDS to serve binary content.
```
```suggestion
Add the file to the following path in your storefront repository. Note the added `.bin` extension, which is currently the only way we have in EDS to serve binary content.
```

bdenham commented on src/content/docs/dropins/payment-services/installation.mdx:
```diff
@@ -3,16 +3,53 @@ title: Payment Services installation
+
+  <Task>
+    ### Register sandbox domain
+    Add a <Link href="https://www.aem.live/docs/redirects" text="redirect" /> to make the domain verification file accessible on https://your-sandbox-domain.com/.well-known/apple-developer-merchantid-domain-association. Once the redirect is in place, contact your sales representative to register the sandbox domain with Apple.
```
```suggestion
Add a <Link href="https://www.aem.live/docs/redirects" text="redirect" /> to make the domain verification file accessible at `https://your-sandbox-domain.com/.well-known/apple-developer-merchantid-domain-association`. Once the redirect is in place, contact your sales representative to register the sandbox domain with Apple.
```

bdenham commented on src/content/docs/dropins/payment-services/installation.mdx:
```diff
@@ -3,16 +3,53 @@ title: Payment Services installation
+    | .well-known/apple-developer-merchantid-domain-association | /.well-known/apple-developer-merchantid-domain-association.bin |
+
+    <Aside type="caution" title="Sandbox only">
+      Serving the file behind a redirect works for sandbox domain registration but not for production. For production, the file must be served directly, without a redirect.
```
```suggestion
Serving the file via a redirect works for sandbox domain registration, but not for production. In production, the file must be served directly, without a redirect.
```

bdenham commented on src/content/docs/dropins/payment-services/installation.mdx:
```diff
@@ -3,16 +3,53 @@ title: Payment Services installation
+
+  <Task>
+    ### Register production domain
+    Set up a <Link href="/setup/configuration/content-delivery-network/#url-rewrites" text="URL rewrite" /> at the CDN level to make the domain verification file directly accessible on https://your-production-domain.com/.well-known/apple-developer-merchantid-domain-association, without redirects. Once the URL rewrite is in place, contact your sales representative to register the production domain ...
```
```suggestion
Set up a <Link href="/setup/configuration/content-delivery-network/#url-rewrites" text="URL rewrite" /> at the CDN level to make the domain verification file directly accessible at https://your-production-domain.com/.well-known/apple-developer-merchantid-domain-association, without redirects. After the rewrite is in place, contact your sales representative to register the production domain with Apple.
```

bdenham reviewed (approved):
Appoved after making the suggested changes.

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/index.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/index.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/index.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/initialization.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/initialization.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/installation.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/installation.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/installation.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/installation.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/installation.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/installation.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/installation.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

AVoskoboinikov commented on src/content/docs/dropins/payment-services/containers/apple-pay.mdx:
```diff
@@ -0,0 +1,45 @@
+    ['getCartId', 'function', 'Yes', 'Should return a promise that resolves to the shopper`s cart ID.'],
+    ['isVirtualCart', 'boolean', 'Yes', 'Whether the cart contains only virtual/downloadable products.'],
+    ['onButtonClick', 'function', 'No', 'Called when the user clicks the Apple Pay button. The callback receives a "showPaymentSheet" function as its only argument that must be called to begin the Apple Pay session and show the payment sheet. If no "onButtonClick" handler is provided, the Apple Pay session will begin and the payment sheet will show on click, always.'],
+    ['onSuccess', 'function', 'No', 'Called when the payment flow finishes successfully. If the function returns a promise, the promise will be awaited before showing marking the payment as "completed successfully" in the Apple Pay sheet. If the promise rejects, the payment will be marked as "failed" in the Apple Pay sheet and the error will be propagated to the "onError" callback, if provided.'] ...
```
> before showing marking the payment as "completed successfully"

are both verbs required here - "showing marking"?

Tomasito665 replied:
sharp! thanks. removed

Tomasito665 committed:
Update src/content/docs/dropins/payment-services/installation.mdx

Co-authored-by: Bruce Denham <bdenham@adobe.com>

Commit:
Remove redundant verb from apple pay 'onSuccess' prop description

Commit:
Move min APS version to Aside note

Commit:
Refactor "Storefront domain registration" tutorial

- Make entire thing PaaS/SaaS agnostic
- Make only specific steps SaaS-specific
- Align badge to the right part of the note

danidelcar reviewed (approved):
reviewed PR today with @Tomasito665
